gen/app-gen

```diff
@@ -208,9 +208,18 @@ if [ -z "${manifests_dir}" ]; then
     show_help_and_exit_error
 fi
 
-if [ ${#images[@]} -lt 1 ]; then
-    echo "No images specified. Aborting." >&2
-    show_help_and_exit_error
+if [ ${#images[@]} -lt 3 ]; then
+    echo "No specific images specified. Will try to extract from docker-compose.yaml file." >&2
+
+    while IFS= read -r img; do
+        images+=("$img")
+        images+=("$img")
+    done <<< $( docker compose --project-directory "$manifests_dir" config --images)
```

Comment:
thank you it is a lovely feature you are proposing.
you do not have to use _here strings_ and play with `IFS`, you can:

```suggestion
while read -r img; do
        images+=("$img")
        images+=("$img")
done < <( docker compose --project-directory "$manifests_dir" config --images)
```

and finally one day you could send the patch to accept "current" and "new" `manifests_dir` -- so you could modify your code to generate deltas based on the manifests!
I like your idea so much, that I would also think about doing the same for the `k8s` case.

Reply:
Thanks for the suggestion, works as intended. Will have to take a look on the mechanism behind `IFS` :+1: